rust_src/src/fileio.rs

```diff
@@ -0,0 +1,18 @@
+//! Functions to deal with files
+
+use lisp::defsubr;
+use multibyte::LispStringRef;
+use remacs_macros::lisp_fn;
+use std::path;
+
+/// Return non-nil if NAME ends with a directory separator character.
+#[lisp_fn]
+pub fn directory_name_p(name: LispStringRef) -> bool {
```

Comment:
The more common style in the tree is:

    if name.len_bytes() <= 0 {
        return false;
    }

    let b = name.byte_at(name.len_bytes() - 1);
    b as char == path::MAIN_SEPARATOR

Reply:
Changed as suggested.

Reply:
No. It doesn't make sense. Changed.

Reply:
:+1: Sorry about that.